fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -280,13 +277,9 @@ public Response deleteObject() {
 
         try {
             resource().delete();
-
-            try {
-                session.save();
-            } catch (final RepositoryException e) {
-                throw new RepositoryRuntimeException(e);
+            if (!batchService.exists(session.getId(), getUserPrincipal())) {
```

Comment:
As stated below, I have rolled back the commits that implement this. There are ways to encapsulate the behavior of batch sessions, but that will be for a separate effort.